subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -53,17 +62,17 @@ static void handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
 	scan_result++;
 
 	if (scan_result == 1) {
-		shell_fprintf(context.shell, SHELL_NORMAL,
-			      "%-4s | %-32s %-5s | %-4s | %-4s | %-5s\n",
-			      "Num", "SSID", "(len)", "Chan", "RSSI", "Sec");
+		print(context.shell, SHELL_NORMAL,
+		      "%-4s | %-32s %-5s | %-4s | %-4s | %-5s\n",
+		      "Num", "SSID", "(len)", "Chan", "RSSI", "Sec");
 	}
 
-	shell_fprintf(context.shell, SHELL_NORMAL,
-		      "%-4d | %-32s %-5u | %-4u | %-4d | %-5s\n",
-		      scan_result, entry->ssid, entry->ssid_length,
-		      entry->channel, entry->rssi,
-		      (entry->security == WIFI_SECURITY_TYPE_PSK ?
-		       "WPA/WPA2" : "Open"));
+	print(context.shell, SHELL_NORMAL,
+	      "%-4d | %-32s %-5u | %-4u | %-4d | %-5s\n",
+	      scan_result, entry->ssid, entry->ssid_length,
+	      entry->channel, entry->rssi,
+	      (entry->security == WIFI_SECURITY_TYPE_PSK ?
+	       "WPA/WPA2" : "Open"));
 }
 
 static void handle_wifi_scan_done(struct net_mgmt_event_callback *cb)
```

Comment:
At least for TI WiFi driver (probably true for other drivers too) scan results are only returned at the behest of the scan command, so in this case, it might be OK to just set `context.shell = shell ` in the cmd_wifi_scan() function, and use shell_fprintf() always.    
But still, for safety, seems the shell should be set by some kind of module init function.